extensions/resource-deployment/src/ui/deployAzureSQLVMWizard/deployAzureSQLVMWizard.ts

```diff
@@ -20,17 +20,23 @@ import { SqlServerSettingsPage } from './pages/sqlServerSettingsPage';
 import { AzureSQLVMSummaryPage } from './pages/summaryPage';
 import { EOL } from 'os';
 import * as nls from 'vscode-nls';
+import { ToolsAndEulaPage } from '../notebookWizard/notebookWizardToolsAndEulaPage';
+import { IResourceTypeService } from '../../services/resourceTypeService';
 const localize = nls.loadMessageBundle();
 
-export class DeployAzureSQLVMWizard extends WizardBase<DeployAzureSQLVMWizard, WizardPageBase<DeployAzureSQLVMWizard>, DeployAzureSQLVMWizardModel> {
+export class DeployAzureSQLVMWizard extends WizardBase<WizardPageBase<DeployAzureSQLVMWizard, DeployAzureSQLVMWizardModel>, DeployAzureSQLVMWizardModel> {
 	private cache: Map<string, any> = new Map();
+	private _wizardInfo!: AzureSQLVMWizardInfo;
 
-	constructor(private wizardInfo: AzureSQLVMWizardInfo, private _notebookService: INotebookService, private _toolsService: IToolsService) {
+	constructor(private _notebookService: INotebookService, private _toolsService: IToolsService, resourceType?: ResourceType, resourceTypeService?: IResourceTypeService) {
 		super(
 			constants.WizardTitle,
 			'DeployAzureSqlVMWizard',
 			new DeployAzureSQLVMWizardModel(),
-			_toolsService
+			_toolsService,
+			false,
+			resourceType!,
```

Comment:
You are telling it that it's defined yet the param can take in an undefined...some options are
* we shouldn't allow undefined here in the first place
* we initialize to a default value if undefined
* the super class can handle undefined

Reply:
It should be a defined parameter. I am making the changes for it